stats/Record.md

```diff
@@ -104,3 +110,11 @@ if (spanContext.getTraceOptions().isSampled()) {
   measurementMap.record(tagContext);
 }
 ```
+
+## Flush
+
+All libraries should provide a Flush function that, once it returns, guarantees that all
```

Comment:
Also add a statement about the fact that this is not recommended to be called be applications (can add overhead) on general use and would be better to leave the library to schedule the flush calls if possible for performance reasons.

Reply:
I assume you mean that the library should install a shutdown hook or similar. In Go there is nothing like "atexit" or ShutdownHook - the application needs to call Flush explicitly before main returns.

In any case, IMHO a library should never install shutdown hooks, even if they are available in the language.

Reply:
No, I want to discourage people to call this randomly (I finished one request let's call flush) and to rely on the library that stats will be exported every X seconds normally.

In some specific cases this may be needed for example when run on AppEngine or Lambdas etc but not for normal applications running in containers/VMs.

Reply:
I don't mean people should call it randomly. I mean it should be called at the end of the program (at the end of the main function).

Reply:
This should be clearly documented. Because based on Google experience people are calling this in their application in random stages.

Reply:
The alternative which would prevent that would be to provide a Shutdown function that actually prevents any traces or stats from being written after it is called (would probably need to log errors if more traces or stats were recorded).